src/vm/i386/cgenx86.cpp

```diff
@@ -303,6 +305,26 @@ void TransitionFrame::UpdateRegDisplayHelper(const PREGDISPLAY pRD, UINT cbStack
     pRD->ControlPC = *PTR_PCODE(pRD->PCTAddr);
     pRD->Esp  = (DWORD)(pRD->PCTAddr + sizeof(TADDR) + cbStackPop);
 
+#ifdef WIN64EXCEPTIONS
+    pRD->IsCallerContextValid = FALSE;
+    pRD->IsCallerSPValid      = FALSE;
+
+    pRD->pCurrentContext->Eip = pRD->ControlPC;
+    pRD->pCurrentContext->Esp = GetSP();
```

Comment:
It seems that stack walker find the target frame based on the address of established frame (which is just Esp at that point).

When I set ``pRD->pCurrentContext->Esp`` as the above adjusted value, stack walker failed to recognize the target frame (which results in assertion failure).